Choice of automatic login user appears too often

StephanStS opened the pull request:
The choice of the user for an automatic logic (with its dialog) appeared too often.  E.g. if you selected "LightDM login mask" the user name choice was displayed (what should not be the case).  
Now the choice is only done in cases of automatic login (dietpi-autologin options 7 and 2).

**Status**: Testing
- [x] Changed automatic login selection 

@MichaIng: Please check whether my changes are correct.

StephanStS committed:
Choice of automatic login user appears too often

The choice of the user for an automatic logic (with its dialog) appeared too often. 
Now the choice is only done in cases of automatic login (dietpi-autologin options 7 and 2).

MichaIng commented:
Many thanks for the report.

> if you selected "LightDM login mask" the user name choice was displayed (what should not be the case).

That is indeed unintentionally the case when no desktop or other X application has been installed yet. If a desktop is installed, the condition above applies and no user choice is shown: https://github.com/MichaIng/DietPi/pull/4144/files#diff-f6589704b6b6390a7a47c21906be7ab7d2a633b368c849befb5cff8122a3480aR117

In all other autologin cases the user choice makes sense, so I'd instead remove the `command -v startx` check, respectively relocate it and better error out when no X server has been installed yet, as the lightdm service would be doomed to fail anyway.
__________
I think that check was added to postpone the heavier package installs to the desktop install that might be done afterwards and the autologin user selection was added afterwards, I guess. However, it's better to be explicit IMO and no allow an autostart selection that cannot work yet.

MichaIng committed:
v7.0

+ DietPi-Autostart | Assure that if no X server has been installed yet, LightDM selection does not trigger an autologin user selection. Instead prompt a failure and inform user that an X server must be installed first.

MichaIng committed:
v7.0

+ DietPi-Software | Do not offer to install and configure No-IP from within the software selection. It will fail as of concurrent dietpi-software execution. Since this was the only reason for writing/reading a temporary install state file, this function can be removed from the code. Instead allow to configure No-IP after it has been installed.
+ DietPi-Software | Offer to configure DietPi-AutoStart after installs instead of before installs. We'd need to allow failing systemctl calls and others to allow this, and need to keep re-configuring DietPi-AutoStart (automated) after installs. It's better do have the ability to for error handling all required steps and hence relying on software services and binaries being available so that we can check and assure that all worked fine. Since DietPi-Software does not force a reboot after installs, this interactive step after installs have finished is pretty fine and it adds to the anyway possible GPU memory split prompt + No-IP config with above change.
